test/common/filesystem/filesystem_impl_test.cc

```diff
@@ -145,6 +148,17 @@ TEST_F(FileSystemImplTest, IllegalPath) {
 #endif
 }
 
+#ifndef WIN32
+TEST_F(FileSystemImplTest, illegalCanonicalPath) {
```

Comment:
Please add a comment that describes this test as checking for paths that should be allowed but may not exist at test time.